pkg/externalresource/manager/service.go

```diff
@@ -54,5 +167,199 @@ func (s *Service) CreateResource(
 		return nil, st.Err()
 	}
 
+	// Note that the cache is updated AFTER the metastore is updated.
+	if _, exists := s.cache[request.GetResourceId()]; exists {
+		log.L().Panic("Cache is inconsistent",
+			zap.String("resource-id", request.GetResourceId()))
+	}
+	s.cache[request.GetResourceId()] = resourceRecord
+
+	// TODO: handle the case where resourceRecord.Deleted == true
 	return &pb.CreateResourceResponse{}, nil
 }
+
+func (s *Service) GetPlacementConstraint(
```

Comment:
Would you please add comments about what this function does?

Reply:
(and the meaning of returned bool, it's true or false for no constraint?)